src/amo/commands/updateTaskbar.ts

```diff
@@ -17,38 +20,44 @@ export async function updateTaskbar() {
     return;
   }
 
-  const path = doc.uri.fsPath;
+  const filePath = doc.uri.fsPath;
   const rootFolder = vscode.workspace.workspaceFolders?.[0].uri.fsPath;
   if (!rootFolder) {
     return;
   }
-  const relativePath = path.replace(rootFolder, "");
-  const pathParts = relativePath.split("/");
-  let guid = pathParts[1];
-  const version = pathParts[2];
 
+  const relativePath = filePath.replace(rootFolder, "");
+  const [guid, version] = relativePath.split(path.sep).splice(-2);
   if (!guid || !version) {
     return;
   }
 
-  if (
-    guid.length === 36 &&
-    guid[8] === "-" &&
-    guid[13] === "-" &&
-    guid[18] === "-" &&
-    guid[23] === "-"
-  ) {
-    // looks like a guid
-    guid = "{" + guid + "}";
-  }
+  const reviewUrl = `${constants.reviewBaseURL}${guid}`;
+  try {
+    const response = await Promise.race([
+      fetch(reviewUrl),
+      new Promise<Response>((_, reject) =>
+        setTimeout(() => reject(new Error("Request timed out")), 2000)
```

Comment:
node-fetch docs suggest using a controller for timeouts, fwiw https://www.npmjs.com/package/node-fetch#request-cancellation-with-abortsignal